lib/taskana-core/src/main/java/pro/taskana/common/internal/jobs/AbstractTaskanaJob.java

```diff
@@ -54,4 +58,17 @@ public static TaskanaJob createFromScheduledJob(
     }
     return result;
   }
+
+  protected Instant getNextDueForCleanupJob() {
+    Instant nextRun = Instant.now();
+    if (scheduledJob != null && scheduledJob.getDue() != null) {
+      nextRun = scheduledJob.getDue();
+    }
+
+    while (nextRun.isBefore(Instant.now())) {
+      nextRun = nextRun.plus(runEvery);
+    }
```

Comment:
Why did you wrap this in a while loop? 
Isn't a simple addition enough?

Reply:
It is a measure to ABSOLUTELY ensure that the problem does not appear. A possible problem could appear if the due of the previous job is far in the past, for whatever reason. 
I also thought about just adding runEvery to the current time, no matter what the due of the last scheduled job has been.

Reply:
> It is a measure to ABSOLUTELY ensure that the problem does not appear. A possible problem could appear if the due of the previous job is far in the past, for whatever reason.
What's the issue here? What will happen when the entry will be set to a date in the past? I am not 100% sure, but I think on the next check that job will just be executed again. Isn't that the behaviour we want? Maybe we can talk with @holgerhagen or @gitgoodjhe about the expected behaviour here.